redisinsight/ui/src/components/database-overview/DatabaseOverview.tsx

```diff
@@ -87,13 +93,17 @@ const DatabaseOverview = () => {
       isBdbPackages,
     } = {},
   } = overview
-
   const loadData = () => {
-    if (connectedInstanceId) {
+    if (connectedInstanceId && !connectivityError) {
       dispatch(getDatabaseConfigInfoAction(connectedInstanceId))
       setLastRefreshTime(Date.now())
     }
   }
+  useEffect(() => {
+    if (!connectivityError) {
+      loadData()
+    }
+  }, [connectivityError])
```

Comment:
I don't like this approach as it can lead to issues and is hard to follow:
loadData - an arrow function that's generated on each render
the useEffect - triggers only when connectivityError changes so it can use stale loadData function (e.g. connectedInstanceId has now changed)

is it intentional that the dispatch shouldn't trigger if the connectedInstanceId changes? If not, a possible refactor could be:

```suggestion
useEffect(() => {
  if (!connectivityError && connectedInstanceId) {
    dispatch(getDatabaseConfigInfoAction(connectedInstanceId))
    setLastRefreshTime(Date.now())
  }
}, [connectivityError, connectedInstanceId])
```

Reply:
`loadData` is used in AutoRefresh component. 
I don't see how it can be stale, as you are pointing out, it is recreated on each render.
The purpose of the effect is to re-trigger autorefresh once connectivity is no longer broken, because when there is no connection and you open the app, you get an error and no metrics. If you get no metrics, auto refresh is not set, and you never get metrics, unles you hard reload the app.

Reply:
it is recreated but the new version will only be used if connectivityError changes; should any of it's other dependancies change (e.g. connectedInstanceId) it won't trigger the useEffect

I suppose in this case this works fine, it's just that this approach is prone to errors as the code grows. See a cool article by Dan Abramov on the topic ([link](https://overreacted.io/a-complete-guide-to-useeffect/#moving-functions-inside-effects))
There's a useful eslint rule that we have turned off for some reason - react-hooks/exhaustive-deps, it forbids such things